src/lib/wapi.js

```diff
@@ -2216,29 +2219,101 @@ window.WAPI.demoteParticipant = function (idGroup, idParticipant, done) {
  * Send Sticker
  * @param {*} sticker 
  * @param {*} chatId '000000000000@c.us'
- * @param {*} done - function - Callback function to for async execution.
  */
-window.WAPI.sendSticker = function (sticker, chatid, done) {
-	var idUser = new window.Store.UserConstructor(chatid, { intentionallyUsePrivateConstructor: true });
-	return window.Store.Chat.find(idUser).then(async (chat) => {
-		// let stick = new window.Store.Sticker.modelClass();		
-		// stick.__x_clientUrl = sticker.url;
-		// stick.__x_filehash = sticker.filehash;
-		// stick.__x_id = sticker.filehash;
-		// stick.__x_uploadhash = sticker.uploadhash;
-		// stick.__x_mediaKey = sticker.mediaKey;
-		// stick.__x_initialized = false;
-		// stick.__x_mediaData.mediaStage = 'INIT';
-		// stick.mimetype = 'image/webp';
-		// stick.height = 512;
-		// stick.width = 512;
-		await sticker.initialize();
-		await sticker.sendToChat(chat);
-		if (done !== undefined) done(true);
-	})
-	.catch(e => {
-		if (done !== undefined) done(false);
-	});
+window.WAPI.sendSticker = async function ({sticker, chatid}) {
+    // console.log(sticker);
+    // console.log(chatid);
+    var chat = Store.Chat.get(chatid)
+    console.log(chat);
+	// var idUser = new window.Store.UserConstructor(chatid, { intentionallyUsePrivateConstructor: true });
+	// return  (chat) => {
+        // ephemeralStartTimestamp
+		let stick = new window.Store.Sticker.modelClass();		
+		stick.__x_clientUrl = sticker.clientUrl;
+		stick.__x_filehash = sticker.filehash;
+		stick.__x_id = sticker.filehash;
+		stick.__x_uploadhash = sticker.uploadhash;
+		stick.__x_mediaKey = sticker.mediaKey;
+		stick.__x_initialized = false;
+		stick.__x_mediaData.mediaStage = 'INIT';
+		stick.mimetype = 'image/webp';
+		stick.height = 512;
+		stick.width = 512;
+        console.log("window.WAPI.getMessageById -> stick", stick)
+		await stick.initialize();
+		return await stick.sendToChat(chat);
+		// if (done !== undefined) done(true);
+	// })
+	// .catch(e => {
+    //     console.log(e)
+	// });
+};
+
+window.WAPI.getFileHash = async (data) => {
+	let buffer = await data.arrayBuffer();
+	var sha = new jsSHA("SHA-256", "ARRAYBUFFER");
+	sha.update(buffer);
+	return sha.getHash("B64");
+};
+
+window.WAPI.generateMediaKey = async (length) => {
+    var result = '';
+    var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+    var charactersLength = characters.length;
+    for ( var i = 0; i < length; i++ ) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    }
+    return result;
+};
+
+window.WAPI.encryptAndUploadFile = async function ({ type, data }) {	
```

Comment:
Needs to use params instead of object to maintain consistency throughout WAPI.js

Reply:
👍